lib/Optimizer/Transforms/DecompositionPatterns.cpp

```diff
@@ -269,6 +269,32 @@ struct SToPhasedRx : public OpRewritePattern<quake::SOp> {
   }
 };
 
+// quake.s target
+// ────────────────────────────────────
+// Gate equivalence:
+// quake.s == quake.r1(π/2, 0)
+// This will enable further decomposition via other patterns such as
+// controlled-r1 to cnot.
+struct SToR1 : public OpRewritePattern<quake::SOp> {
+  using OpRewritePattern<quake::SOp>::OpRewritePattern;
+
+  void initialize() { setDebugName("SToR1"); }
+
+  LogicalResult matchAndRewrite(quake::SOp op,
+                                PatternRewriter &rewriter) const override {
+    if (!quake::isAllReferences(op))
+      return failure();
```

Comment:
Not for this PR: it seems like this could easily allow wires as inputs. There is no change at all to the semantics, of course. The only difference would be creating the new op with an identical set of return types.